azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/core/AbstractCredentialRetriever.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.tools.auth.core;
+
+import com.microsoft.azure.AzureEnvironment;
+import com.microsoft.azure.tools.auth.exception.LoginFailureException;
+import com.microsoft.azure.tools.auth.model.AzureCredentialWrapper;
+import lombok.Setter;
+import rx.Single;
+
+public abstract class AbstractCredentialRetriever implements ICredentialRetriever {
+
+    @Setter
+    protected AzureEnvironment env;
+
+    public Single<AzureCredentialWrapper> retrieve() {
```

Comment:
do we have `AzureCredential`? what is the difference between `AzureCredentialWrapper` and `AzureCredential`? and why we need a `Wrapper`?

Reply:
AzureCredentialWrapper is a wrapper of new auth model: TokenCredential, we need a wrapper for authMethod and defaultSubscriptionId